app/models/user_mailbox.rb

```diff
@@ -0,0 +1,43 @@
+class UserMailbox
+  attr_reader :user
+
+  def initialize(user)
+    @user = user
+  end
+
+  def inbox
+    messages = user.mailbox.inbox
+    messages.each{|m| m.mark_as_read(user)}
+  end
+
+  def delete_all
+    user.mailbox.inbox.each do |msg|
+      delete_message(msg)
+    end
+    empty_trash(user)
+  end
+
+  def destroy message_id
+    error_msg = nil
+    msg = Mailboxer::Conversation.find(message_id)
+    if (msg.participants[0] == user) || (msg.participants[1] == user)
+      delete_message(msg)
+      empty_trash(msg.participants[0])
+    else
+      error_msg = "You do not have privileges to delete the notification..."
+    end
+    error_msg
+  end
+
+  private
+
+  def delete_message (msg)
+    msg.move_to_trash(msg.participants[0])
+    msg.move_to_trash(msg.participants[1])
+  end
+
+  def empty_trash (user)
```

Comment:
Rubocop says: no spaces between a method name and its parenthesized params in a method definition.